tests/test_generate_ics.py

```
@@ -418,12 +456,55 @@ def test_filename_year_offset(self):
+            self.assertTrue((temp_dir / "US_20250701.ics").exists())
+
+    def test_output_template_with_braces(self):
+        with self.temp_cwd() as temp_dir:
+            with self.argv("US", "--years", "2025", "--output-template", "{{{code}}}.ics"):
+                IcsGenerator().run()
+
+            self.assertTrue((temp_dir / "{US}.ics").exists())
+
+    def test_output_template_unknown_placeholder(self):
+        with self.argv("US", "--output-template", "{foo}.ics"):
+            with self.assertRaises(SystemExit) as context:
```

Ruff (0.15.20)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)